docs/probe.md

```diff
@@ -0,0 +1,177 @@
+# Probes and Meters: Examples for Adding to Code
+For an introduction to `Probe`s and `Meter`s, please refer to [Measurement Overview](./metrics.md).
+
+## Context
```

Comment:
do we discuss at all where to find the outputted results of what we're measuring?

Reply:
by default, output will go to the `json` file, but there's a separate issue where we run in to record size restrictions such as:
> 2022-12-16 19:34:36 30s WARNING  armory.instrument.instrument:_write:856 record (name=my_layer_identity, batch=0, result=...) size > max_record_size 1048576. Dropping.
which will depend on what the user is trying to capture. It just so happens that the examples shown here do run into those issues, but that should be its own section about defining a new class of `Meter` to:
1. work around the size limitation
2. and save to a `pkl` file rather than `json`

Reply:
that said, I guess we should just mention that outputs will go to the `json` file while we figure out a better alternative?

Reply:
check out f402f30d